lib/PuppeteerSharp.Tests/PuppeteerTests/PuppeteerLaunchTests.cs

```diff
@@ -281,5 +282,63 @@ public void ShouldReturnTheDefaultChromeArguments()
                 Assert.Equal("Unable to create or connect to another chromium process", exception.Message);
             }
         }
+
+        [Fact]
+        public void ShouldDumpBrowserProcessStderr()
+        {
+            var dumpioTextToLog = "MAGIC_DUMPIO_TEST";
+            var success = false;
+            var process = new DirectoryInfo(Directory.GetCurrentDirectory()).Name.Contains("netcore") ?
+                GetDumpIOCoreProcess(dumpioTextToLog) :
+                GetDumpIOFrameworkProcess(dumpioTextToLog);
+
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.RedirectStandardError = true;
+
+            process.ErrorDataReceived += (sender, e) =>
+            {
+                success |= e.Data != null && e.Data.Contains(dumpioTextToLog);
+            };
+
+            process.Start();
+            process.BeginErrorReadLine();
+            process.WaitForExit();
+            Assert.True(success);
+        }
+
+        private Process GetDumpIOFrameworkProcess(string dumpioTextToLog)
+        {
+            var process = new Process();
+            process.StartInfo.FileName = Path.Combine(
+                GetDumpIOAppDirectory(new DirectoryInfo(Directory.GetCurrentDirectory()).Parent),
+                "PuppeteerSharp.Tests.DumpIO.exe");
+            process.StartInfo.Arguments = $"{dumpioTextToLog} " +
+                $"\"{new BrowserFetcher().RevisionInfo(BrowserFetcher.DefaultRevision).ExecutablePath}\"";
+
+            return process;
+        }
+
+        private Process GetDumpIOCoreProcess(string dumpioTextToLog)
+        {
+            var process = new Process();
+            process.StartInfo.WorkingDirectory = GetDumpIOAppDirectory(new DirectoryInfo(Directory.GetCurrentDirectory()));
+            process.StartInfo.FileName = "dotnet";
+            process.StartInfo.Arguments = $"PuppeteerSharp.Tests.DumpIO.dll {dumpioTextToLog} " +
+                $"\"{new BrowserFetcher().RevisionInfo(BrowserFetcher.DefaultRevision).ExecutablePath}\"";
+
+            return process;
+        }
+
+        private string GetDumpIOAppDirectory(DirectoryInfo baseDir)
```

Comment:
maybe instead of relying on the `baseDir` we could extract the version from the dll itself or something like this https://github.com/cake-build/cake/blob/develop/src/Cake.Core/Polyfill/EnvironmentHelper.cs#L119